### PR TITLE
AggregateCoercionNode  bug.

### PR DESCRIPTION
**Pull request by MikeStall**
We only add fields to AggregateCoercionNode that require a coercion. But eval only evals the fields listed in AggregateCoercionNode. So a field that doesn't require coercion gets skipped in eval!

This fixes it via eval - but should probalby be fixed at IR level. Since eval processes every time, whereas we know this statically.

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/ArgCoercion.txt [hunk 1]
```diff
@@ -40,4 +40,16 @@ Error({Kind:ErrorKind.InvalidArgument})
...
+// Ensure record coercion doesn't prevent us from accessing non-coerced fields. 
+>> Last([{a:1,b:2}, {a:"1", b: 3}]).b
+3
+
+// extra nesting, a.x coerces. 
+>> Last([{a: { x:1},b:2}, {a:{x:"1"}, b: 3}]).b
+3
+
+// Coerce to 1st element in list; so middle element shouldn't impact  
+>> Last([{a:1,b:2}, {a:2, b: "2"}, {a:"1", b: 3}]).b
```
Comment (Shpakh-MSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/ArgCoercion.txt:54]: Last([ {a:2}, {a:true, b:true, c:20}, {a:2, b:2, c:true} ]).c

This should return true.

**Review by Shpakh-MSFT — APPROVED** [review]
:shipit:

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/ArgCoercion.txt (hunk 1 above)
Comment (anderson-joyle) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/ArgCoercion.txt:54]: The first record `{a:1,b:2}` "initializes" field `b` as number.

Comment (MikeStall) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/ArgCoercion.txt:54]: added

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by anderson-joyle — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**